### PR TITLE
refactor: Convert Form stories

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Convert Label, HelpText, TextField, and TextFieldSearch stories
- Add Input story

## **Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 46 57
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 46 58
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 47 05
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 47 13
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 51 22
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 51 25
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 51 41
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 51 46
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 51 57
Simulator Screen Shot - iPhone 14 Pro - 2023-10-02 at 08 51 58

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Form/HelpText/HelpText.constants.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const TEXT_COLOR_BY_HELPTEXT_SEVERITY: TextColorByHelpTextSeverity = {
 
 // Sample consts
 export const SAMPLE_HELPTEXT_TEXT = 'Sample HelpText Text';
```
Comment (georgewrmarshall) [inline app/component-library/components/Form/HelpText/HelpText.constants.ts:28]: nit: Our content guidelines suggest we use sentence case for everything. We should reenforce this in our documentation even with sample text 
```suggestion
export const SAMPLE_HELPTEXT_TEXT = 'Sample HelpText text';
```

File: app/component-library/components/Form/TextField/foundation/Input/Input.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ const Input: React.FC<InputProps> = ({
   ...props
 }) => {
   const [isFocused, setIsFocused] = useState(autoFocus);
-
+  console.log(isStateStylesDisabled);
```
Comment (georgewrmarshall) [inline app/component-library/components/Form/TextField/foundation/Input/Input.tsx:28]: Should this be removed?

File: app/component-library/components/Form/TextFieldSearch/TextFieldSearch.constants.ts [hunk 3]
```diff
@@ -22,4 +23,9 @@ export const SAMPLE_TEXTFIELDSEARCH_PROPS: TextFieldSearchProps = {
   onPressClearButton: () => {
     console.log('clicked');
   },
+  size: DEFAULT_TEXTFIELD_SIZE,
+  isError: false,
+  isDisabled: false,
+  isReadonly: false,
+  placeholder: 'Sample Placeholder',
```
Comment (georgewrmarshall) [inline app/component-library/components/Form/TextFieldSearch/TextFieldSearch.constants.ts:30]: nit: use sentence case
```suggestion
  placeholder: 'Sample placeholder',
```

File: app/component-library/components/Form/TextFieldSearch/TextFieldSearch.stories.tsx [hunk 4]
```diff
@@ -1,34 +1,46 @@
...
-  <TextFieldSearch {...getTextFieldSearchStoryProps()} />
-);
-
-storiesOf('Component Library / Form', module).add(
-  'TextFieldSearch',
-  TextFieldSearchStory,
-);
-
-export default TextFieldSearchStory;
+export const TextFieldSearch = {};
```
Comment (georgewrmarshall) [inline app/component-library/components/Form/TextFieldSearch/TextFieldSearch.stories.tsx:46]: I'm experiencing an issue with `showClearButton` enabled and being able to edit the TextFieldSearch


https://github.com/MetaMask/metamask-mobile/assets/8112138/7034483c-7a2c-4395-8e39-f4f30f1160e7

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! A couple comments. Also one suggestion to make PRs easier to read I would suggest using screen casts instead of screenshots e.g.


https://github.com/MetaMask/metamask-mobile/assets/8112138/566a8032-bd3e-4c39-9d97-e4506bf6c56b

File: app/component-library/components/Form/TextFieldSearch/TextFieldSearch.stories.tsx (hunk 4 above)
Comment (brianacnguyen) [reply app/component-library/components/Form/TextFieldSearch/TextFieldSearch.stories.tsx:44]: It seems to be working for me. Can you double check?
Simulator Screen Shot - iPhone 14 Pro - 2023-10-25 at 21 20 06

**Review by georgewrmarshall — APPROVED** [review]
TextFieldSearch working for me now. Must have been an issue on my side only. LGTM

**Pull request merged**